Feat/6 format titles

arunge opened the pull request:
## Version 24.03.3

### New Features
- module for plotExport (#6)
  - option to add and format titles of a plot before export if the plot is a ggplot (1)
  - option to set ranges of a plot before export if the plot is a ggplot (2)

![image](https://github.com/Pandora-IsoMemo/shiny-tools/assets/16759098/c3148e89-0cbe-4bc7-9c9d-139ce8af539f)

f-lukas commented on R/02-plotTitles.R:
```diff
@@ -0,0 +1,223 @@
+                 }) %>%
+                   bindEvent(input[["labelName"]])
+
+                 observe({
```
Just an idea because most functions are quite long. We could move all of the observer from line 103 to 131 into a wrapper function (with a meaningful name). We could then only call this function here which would improve readability a lot. Maybe this can also be done for other modules.

f-lukas reviewed (approved):
The functionality is really nice and almost everything worked right away :+1: I only had problems with the pdf export but I guess this is not a problem with the code but maybe I have to install some fonts or anything. But we should test if it works when deployed.

![Bildschirmfoto vom 2024-03-20 08-48-30](https://github.com/Pandora-IsoMemo/shiny-tools/assets/86153843/02699eb6-5896-4038-877b-f1cf2eb3e6c0)

I only have a smaller note about the code. Some modules contain a lot of stuff which makes it difficult to understand what exactly is happening. I wrote one suggestions as a comment (move code into a wrapper function). I think in some modules it is better to leave the code as is but maybe we could add some comments that separate different things within a module from each other?

arunge commented:
- I extracted the observers
- I added the option to change the style of axis texts also ( I need this in several apps)

I am merging when checks will finish
